jwst/tweakreg/tweakreg_step.py

```diff
@@ -116,12 +119,10 @@ def process(self, input):
                               .format(catalog_filename))
                 image_model.meta.tweakreg_catalog = catalog_filename
 
+            # Temporarily attach catalog to the image model so that it follows
+            # the grouping by exposure, to be removed after use below
             image_model.catalog = catalog
```

Comment:
Hol up: The assignment is still occurring.

Reply:
Right, but it is then being deleted from the model (and asdf tree) before being returned by the step further down.  I.e. before any possibility of it being written out or passed along to the next step.

Unless I'm missing something else here.

It needs to be attached because the items in the container are grouped by exposure, not by file, but each catalog follows each file.  This is another case where we need `ModelContainer` to group things in a certain way and have it be consistent across multiple items.

Clearly we need to investigate a bit further and think of a more elegant way to do this in `ModelContainer`, or make sure attaching to the datamodel does not blow up everything else.

Agree that the `asdf` root cause should probably get solved first before we can consider #6039 solved.